graphql_compiler/compiler/validation.py

```diff
@@ -20,7 +23,7 @@ def validate_schema_and_query_ast(schema, query_ast):
     Returns:
         list containing schema and/or query validation errors
     """
-    core_graphql_errors = validate(schema, query_ast)
+    core_graphql_errors = [str(error) for error in validate(schema, query_ast)]
```

Comment:
It makes more sense to me to be consistent with the GraphQL core next library and have the `validate` function return a list of `GraphQLError` objects. Thoughts?

Reply:
This is a type-hinting PR, so I tried to keep functionality the same. I'd prefer to keep type hinting separate from functionality changes, otherwise the type hinting process becomes super tedious.

We have working, well-tested code in the codebase, even though not all parts are designed in the way we'd design them if we started from scratch today. But if we stop and fix everything that we'd like to redesign *while* we're type-hinting the codebase, we'll never finish type-hinting because there will always be something to fix and redesign.

Let's keep our eyes on the ball here: we're trying to change how people query their data systems. Whether this internal function returns strings, or returns error objects which immediately get stringified by the caller, is completely immaterial for that goal.